Assignment9.txt

```diff
@@ -0,0 +1,31 @@
+OrganisationController:
+In the project in IOrganisationDao there are 8 methods which is implemented in the Organisation class for different purposes.
+In the OrganisationController class we are Controlling every method means there we have passed some URL for each and every method through which we can get our desird result according to our requirement.
+I am explaining the methods one by one
+1.public Organisation save(Organisation organisation):
+This method included KeyHolder interface
+Here we also used GeneratedKeyHolder class.
+Both  are predefined .
+we have created one object of GeneratedKeyHolder calss in save method to get the Key.
+we have created the object of MapSqlParameterSource And we have added the Organisation name and status through the help of the object and the method addValue() which is inside the MapSqlParameterSource Class.
+Here we have used NamedParameterJdbcTemplate .by Autowiring we have created the object and by the help of this and update method we have updated the value.
+QueryOrganisation is a class where we have written the queries for Organisation class.Here We have written many parameters .
+But in save() we used INSERT_ORG.
+INSERT_ORG is used for inserting Organisation name and status And WE have taken two parameters ON and Os respectively.
+Here we have used KeyHolder interface which helps in retrieving keys
+This is used for auto generated keys which potentially return by JDBC insert statement.
+And finally it will return the values Of class type and we can get the value from the entity class.
+2.public List<Role> getAssignedRolesById(Organisation org):
+In this method we will give the Id of Organisation then we will retrieve the values through the parameter GET_ASSIGNED_ROLE_BY_ID.
+3.public List<Role> getNotAssignedRolesById(Organisation org):
+This method is  like as public List<Role> getAssignedRolesById(Organisation org).The parameter name is GET_NOT_ASSIGNED_ROLE_BY_ID.
+4.public Organisation assignMoreRolesById(Organisation org):

```

Comment:
Suppose an organisation has 2 roles. Then we want to add more roles. In that case we will assign more roles. It will be inserted into another table using batchInsert. Look at the query used in this method to know what is the table it is inserting.